React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+            └── button.ts
+```
+
+<!-- TODO maybe detail description of each folder or list of files per folder? -->
+
+### Colors
+
+Color naming should a single value or an object with keys in range from 50 to 900.
+
+Example:
+
+```ts
+export const colors = {
+  white: "#ffffff",
+  primary: {
+    50: "#f6faff",
+    100: "#d7e9fd",
+    200: "#c0dcfc",
+    300: "#a7cefb",
+    400: "#8abdfa",
+    500: "#68aaf8",
+    600: "#3d92f6",
+    700: "#0070f3",
+    800: "#0064d8",
+    900: "#0055b9",
+  },
+};
+```
+
+### Global styles
+
+Global styles are theme-aware styles you can apply to any html element globally.
+
+They are defined in `src/styles/theme/styles.ts` file.
+
+<!-- TODO maybe what should be defined in global -->
+
+Example:
+
+```ts
+const overrides = {
+  styles: {
+    global: {
+      body: {
+        fontFamily: "body",
+        bg: "white",
+        color: "primary.200",
+      },
+      "*": {
+        boxSizing: "border-box",
+      },
+    },
+  },
+};
+```
+
+### Responsive
+
+Chakra UI supports responsive styles out of the box. Instead of manually adding @media queries and adding nested styles throughout your code, Chakra UI allows you to provide object and array values to add mobile-first responsive styles.
+
+> Under the hood `@media(min-width)` media query is used to ensure mobile-first
+
+Chakra UI default breakpoints:
+
+```ts
+export const breakpoints = {
+  sm: "30em",
+  md: "48em",
+  lg: "62em",
+  xl: "80em",
+};
+```
+
+Here's how to interpret this syntax:
+
+- `base`: From `0em` upwards
+- `md`: From `48em` upwards
+- `lg`: From `62em` upwards
+- `xl`: From `80em` upwards
+
+For responsive styles, array or object syntax can be used.
+
+```tsx
+// array syntax
+<Text fontSize={["24px", "40px", "56px"]}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+```tsx
+// object syntax
+<Text fontSize={{ base: "24px", md: "40px", lg: "56px" }}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+In case we need to skip cretan breakpoint, `null` is passed at that position in the array to avoid generating unnecessary CSS.
+
+Example:
+
+```jsx
+<Text fontSize={["24px", null, "56px"]}>Lorem Ipsum is simply dummy text</Text>
+```
+
+Array and Object syntax work for every style prop in the them specification, which means you can change most properties at a given breakpoint.
+
+To create custom breakpoints read [docs](https://chakra-ui.com/docs/features/responsive-styles#customizing-breakpoints).
+
+### Custom components style
+
+Chakra UI has a specific API for styling components. Most components have default or base styles `baseStyle`, styles for different sizes `sizes`, and styles for different visual variants `variants`.
+
+Components styles are defined in `theme/components/<component-name>.ts`
+Each component style will export these objects:
+
+- `baseStyles`
+- `sizes`
+- `variants`
+- `defaultProps`
+
+Base style are styles that all button types share.
+
+```tsx
+const baseStyle = {
```

Comment:
How would you ensure that there are no typos in this object? For example: `lineHeigth` instead of `lineHeight`? Is there a type for this object?

Also how would you ensure that only valid values are used? For example: `cursor: "foo"`

Reply:
> How would you ensure that there are no typos in this object? For example: lineHeigth instead of lineHeight? Is there a type for this object?
I added `StyleObjectOrFn` type for that

> Also how would you ensure that only valid values are used? For example: cursor: "foo"
This is a problem because chakra doesn't have a type for that and it would be hard to create one because the value can be some css value or something from a theme or array (cc @isBatak)